### PR TITLE
Implement missing Nether Blocks

### PR DESCRIPTION
**Pull request by pandaaaBE**
## Introduction
This plugin adds follow things:
- Crimson Nylium/Fungus, Waped Nylium/Fungus, Nether Sporuts
- Bone Meal Logic for Netherrack
- Adds support for Nylium to Mushrooms
- Adds support for Nylium to NetherRoots

Note: Bone Meal logic for Fungus is currently missing and is not planned in this pull request

Project Card: https://github.com/pmmp/PocketMine-MP/projects/14#card-84464878

(reopened due of unresolved conflicts of the old pr)

### API changes
`VanillaBlocks::CRIMSON_NYLIUM(), VanillaBlocks::WARPED_NYLIUM(), VanillaBlocks::CRIMSON_FUNGUS(), VanillaBlocks::WARPED_FUNGUS(), VanillaBlocks::NETHER_SPROUTS()`

## Tests

I tested this PR by doing the following (tick all that apply):
- [x] Writing PHPUnit tests (commit these in the `tests/phpunit` folder)
- [x] Playtesting using a Minecraft client (provide screenshots or a video)
grafik

### CODE REVIEW

File: src/block/Fungus.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	public function onNearbyBlockChange() : void{
+		if($this->getSide(Facing::DOWN)->isTransparent()){
+			$this->position->getWorld()->useBreakOn($this->position);
+		}
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		$down = $this->getSide(Facing::DOWN);
+		$downId = $down->getTypeId();
+		if((!$down->isTransparent()) || $downId === BlockTypeIds::MYCELIUM || $downId === BlockTypeIds::PODZOL || $downId === BlockTypeIds::CRIMSON_NYLIUM || $downId === BlockTypeIds::WARPED_NYLIUM || $downId === BlockTypeIds::FLOWER_POT){
```
Comment (ShockedPlot7560) [inline src/block/Fungus.php]: Unnecessary parentheses
```suggestion
		if(!$down->isTransparent() || $downId === BlockTypeIds::MYCELIUM || $downId === BlockTypeIds::PODZOL || $downId === BlockTypeIds::CRIMSON_NYLIUM || $downId === BlockTypeIds::WARPED_NYLIUM || $downId === BlockTypeIds::FLOWER_POT){
```

Comment (ShockedPlot7560) [reply src/block/Fungus.php]: Would not be better to create a tag for that ?

File: src/block/NetherSprouts.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+class NetherSprouts extends Flowable{
+
+	public function ticksRandomly() : bool{
+		return true;
+	}
+
+	public function onNearbyBlockChange() : void{
+		if($this->getSide(Facing::DOWN)->isTransparent()){
+			$this->position->world->useBreakOn($this->position);
```
Comment (ShockedPlot7560) [inline src/block/NetherSprouts.php]: Use getWorld()

File: src/block/NetherSprouts.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			return [$this->asItem()];
+		}
+
+		return [];
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		$down = $this->getSide(Facing::DOWN);
+		$downId = $down->getTypeId();
+		if((!$down->isTransparent()) || $downId === BlockTypeIds::MYCELIUM || $downId === BlockTypeIds::PODZOL || $downId === BlockTypeIds::CRIMSON_NYLIUM || $downId === BlockTypeIds::WARPED_NYLIUM){
```
Comment (ShockedPlot7560) [inline src/block/NetherSprouts.php]: Same here

File: src/block/Netherrack.php [hunk 4]
```diff
@@ -23,9 +23,41 @@
...
+			return true;
+		}
+
+		return false;
+	}
+
+	private function transformToNylium() : bool{
+		foreach($this->getHorizontalSides() as $side){
+			if($side instanceof Nylium){
+				$this->position->world->setBlock($this->position, $side);
```
Comment (ShockedPlot7560) [inline src/block/Netherrack.php]: Use getWorld ?

File: src/block/Nylium.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+	public function ticksRandomly() : bool{
+		return true;
+	}
+
+	public function onRandomTick() : void{
+		$blockAbove = $this->getSide(Facing::UP);
+		if(!$blockAbove->isTransparent() && $blockAbove->getTypeId() === BlockTypeIds::SNOW_LAYER){
+			//nylium dies
+			BlockEventHelper::spread($this, VanillaBlocks::NETHERRACK(), $this);
```
Comment (ShockedPlot7560) [inline src/block/Nylium.php:55]: Is this really a block spread ?

File: src/block/RedMushroom.php [hunk 6]
```diff
@@ -46,8 +46,7 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 		$position = $this->position;
 		$lightLevel = $position->getWorld()->getFullLightAt($position->x, $position->y, $position->z);
 		$downId = $down->getTypeId();
-		//TODO: nylium support
-		if(($lightLevel <= 12 && !$down->isTransparent()) || $downId === BlockTypeIds::MYCELIUM || $downId === BlockTypeIds::PODZOL){
+		if(($lightLevel <= 12 && !$down->isTransparent()) || $downId === BlockTypeIds::MYCELIUM || $downId === BlockTypeIds::PODZOL || $downId === BlockTypeIds::CRIMSON_NYLIUM || $downId === BlockTypeIds::WARPED_NYLIUM){
```
Comment (ShockedPlot7560) [inline src/block/RedMushroom.php]: Same here for the tag

File: src/block/Fungus.php (hunk 1 above)
Comment (pandaaaBE) [reply src/block/Fungus.php]: You mean something like in NetherRoots?

File: src/block/Nylium.php (hunk 5 above)
Comment (pandaaaBE) [reply src/block/Nylium.php:55]: > Similar to [grass blocks](https://minecraft.wiki/w/Grass_block) and [mycelium](https://minecraft.wiki/w/Mycelium), nylium decays (based on [random ticks](https://minecraft.wiki/w/Random_tick)) into netherrack

Should I use BlockEventHelper::die()?

File: src/block/Fungus.php (hunk 1 above)
Comment (ShockedPlot7560) [reply src/block/Fungus.php]: Like mud or dirt to detect which block can make an exception for a "mushroom" like block

File: src/block/Nylium.php (hunk 5 above)
Comment (ShockedPlot7560) [reply src/block/Nylium.php:55]: Die isn't the right event, its use litteral for coral etc ...
Its not really a spread or a form. I don't see any created event which can handle it.

Comment (pandaaaBE) [reply src/block/Nylium.php:55]: Should I just replace it without a event?

Comment (ShockedPlot7560) [reply src/block/Nylium.php:55]: We need this event, but in other way, Nylium is dying, however it can be confusing

Comment (pandaaaBE) [reply src/block/Nylium.php:55]: Should I add a function called decay()?

File: src/block/Fungus.php (hunk 1 above)
Comment (pandaaaBE) [reply src/block/Fungus.php]: Okay when I am home, I will look at this

File: src/block/Nylium.php (hunk 5 above)
Comment (ShockedPlot7560) [reply src/block/Nylium.php:55]: Okay, so, if we reproduce the logics of grass or mycelium. We need to call a spread event. Sry

Comment (pandaaaBE) [reply src/block/Nylium.php:55]: Alright xD

File: src/block/Fungus.php (hunk 1 above)
Comment (pandaaaBE) [reply src/block/Fungus.php]: @ShockedPlot7560 

now i use the "tags", but i don't know if that's what you wanted.

File: src/block/Fungus.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+class Fungus extends Flowable{
+	use StaticSupportTrait;
+
+	private function canBeSupportedAt(Block $block) : bool{
+		//TODO: moss
+		$supportBlock = $block->getSide(Facing::DOWN);
+		return
+			$supportBlock->hasTypeTag(BlockTypeTags::DIRT) ||
+			$supportBlock->hasTypeTag(BlockTypeTags::MUD) ||
+			$supportBlock->hasTypeTag(BlockTypeTags::NYLIUM);
```
Comment (ShockedPlot7560) [inline src/block/Fungus.php]: Missing soul soil

File: src/block/VanillaBlocks.php [hunk 8]
```diff
@@ -960,13 +965,23 @@ public function getBreakTime(Item $item) : float{
...
+
 		self::register("nether_portal", new NetherPortal(new BID(Ids::NETHER_PORTAL), "Nether Portal", new Info(BreakInfo::indestructible(0.0))));
 		self::register("nether_reactor_core", new NetherReactor(new BID(Ids::NETHER_REACTOR_CORE), "Nether Reactor Core", new Info(BreakInfo::pickaxe(3.0, ToolTier::WOOD))));
 		self::register("nether_wart_block", new Opaque(new BID(Ids::NETHER_WART_BLOCK), "Nether Wart Block", new Info(new BreakInfo(1.0, ToolType::HOE))));
 		self::register("nether_wart", new NetherWartPlant(new BID(Ids::NETHER_WART), "Nether Wart", new Info(BreakInfo::instant())));
 		self::register("netherrack", new Netherrack(new BID(Ids::NETHERRACK), "Netherrack", new Info(BreakInfo::pickaxe(0.4, ToolTier::WOOD))));
 		self::register("note_block", new Note(new BID(Ids::NOTE_BLOCK, TileNote::class), "Note Block", new Info(BreakInfo::axe(0.8))));
-		self::register("obsidian", new Opaque(new BID(Ids::OBSIDIAN), "Obsidian", new Info(BreakInfo::pickaxe(35.0 /* 50 in PC */,  ToolTier::DIAMOND, 6000.0))));
+		self::register("obsidian", new Opaque(new BID(Ids::OBSIDIAN), "Obsidian", new Info(BreakInfo::pickaxe(35.0 /* 50 in PC */, ToolTier::DIAMOND, 6000.0))));
```
Comment (ShockedPlot7560) [inline src/block/VanillaBlocks.php]: Unrelated change

File: src/block/VanillaBlocks.php [hunk 9]
```diff
@@ -1317,8 +1332,7 @@ private static function registerElements() : void{
 		$instaBreak = new Info(BreakInfo::instant());
 		self::register("element_zero", new Opaque(new BID(Ids::ELEMENT_ZERO), "???", $instaBreak));
 
-		$register = fn(string $name, int $id, string $displayName, string $symbol, int $atomicWeight, int $group) =>
-			self::register("element_$name", new Element(new BID($id), $displayName, $instaBreak, $symbol, $atomicWeight, $group));
+		$register = fn(string $name, int $id, string $displayName, string $symbol, int $atomicWeight, int $group) => self::register("element_$name", new Element(new BID($id), $displayName, $instaBreak, $symbol, $atomicWeight, $group));
```
Comment (ShockedPlot7560) [inline src/block/VanillaBlocks.php]: Unrelated change

File: src/block/VanillaBlocks.php [hunk 10]
```diff
@@ -1545,7 +1559,7 @@ public function getLightLevel() : int{ return 15; }
 
 		self::register("warped_wart_block", new Opaque(new BID(Ids::WARPED_WART_BLOCK), "Warped Wart Block", new Info(new BreakInfo(1.0, ToolType::HOE))));
 		self::register("crying_obsidian", new class(new BID(Ids::CRYING_OBSIDIAN), "Crying Obsidian", new Info(BreakInfo::pickaxe(35.0 /* 50 in Java */, ToolTier::DIAMOND, 6000.0))) extends Opaque{
-			public function getLightLevel() : int{ return 10;}
+			public function getLightLevel() : int{ return 10; }
```
Comment (ShockedPlot7560) [inline src/block/VanillaBlocks.php]: Unrelated change

File: src/block/NetherSprouts.php [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+	use StaticSupportTrait;
+
+	private function canBeSupportedAt(Block $block) : bool{
+		//TODO: moss
+		$supportBlock = $block->getSide(Facing::DOWN);
+		return
+			$supportBlock->hasTypeTag(BlockTypeTags::DIRT) ||
+			$supportBlock->hasTypeTag(BlockTypeTags::MUD) ||
+			$supportBlock->hasTypeTag(BlockTypeTags::NYLIUM);
+	}
```
Comment (ShockedPlot7560) [inline src/block/NetherSprouts.php]: Missing soul soil

File: src/block/Nylium.php [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+			$arr = [
+				VanillaBlocks::WARPED_FUNGUS(),
+				VanillaBlocks::NETHER_SPROUTS(),
+				$roots = VanillaBlocks::WARPED_ROOTS(),
+				$roots,
+				$roots
+			];
+		}
+
+		$random = new Random(mt_rand());
```
Comment (ShockedPlot7560) [inline src/block/Nylium.php]: No need to create a random seed

File: src/block/RedMushroom.php
Comment (ShockedPlot7560) [inline src/block/RedMushroom.php:1]: Apply only the minimal changes, there is many unrelated changes

File: src/block/Nylium.php [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+		if($this->getTypeId() === BlockTypeIds::WARPED_NYLIUM){
+			$arr = [
+				VanillaBlocks::WARPED_FUNGUS(),
+				VanillaBlocks::NETHER_SPROUTS(),
+				$roots = VanillaBlocks::WARPED_ROOTS(),
+				$roots,
+				$roots
+			];
+		}
```
Comment (ShockedPlot7560) [inline src/block/Nylium.php:96]: The blocks which will be placed is link to the actual biome, not really the block with which you interact

File: src/block/Nylium.php (hunk 12 above)
Comment (pandaaaBE) [reply src/block/Nylium.php]: to be honest, the code was taken from grass in exactly the same way...

Comment (ShockedPlot7560) [reply src/block/Nylium.php]: Is there a reason why it was done like this ?

File: src/block/RedMushroom.php
Comment (pandaaaBE) [reply src/block/RedMushroom.php:1]: yeah I reformated the code in vanillablocks but why shouldn't Mushrooms use this Trait too?

Comment (ShockedPlot7560) [reply src/block/RedMushroom.php:1]: I dont know, but thats an issue for another PR, not for the moment

Comment (pandaaaBE) [reply src/block/RedMushroom.php:1]: I could rename this pr to "Implement some missing nether blocks and edit RedMushroom to use the Trait"

Comment (ShockedPlot7560) [reply src/block/RedMushroom.php:1]: No, a PR need to done one thing by one thing

File: src/block/NetherSprouts.php (hunk 11 above)
Comment (pandaaaBE) [reply src/block/NetherSprouts.php]: Couldn´t NetherSprouts extends to NetherRoots so I can delete this class?

Comment (pandaaaBE) [reply src/block/NetherSprouts.php]: You already opened a pr to add soul soil so this would not be relevant anymore

Comment (pandaaaBE) [reply src/block/NetherSprouts.php]: nvm fungus could extend to netherroots but sprouts can not be placed in flower pods

File: src/block/Nylium.php (hunk 13 above)
Comment (pandaaaBE) [reply src/block/Nylium.php:96]: What do you think? I just tried it again in Minecraft and I think the behavior is identical.

File: src/block/Fungus.php [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+	use StaticSupportTrait;
+
+	private function canBeSupportedAt(Block $block) : bool{
+		//TODO: moss
+		$supportBlock = $block->getSide(Facing::DOWN);
+		return
+			$supportBlock->hasTypeTag(BlockTypeTags::DIRT) ||
+			$supportBlock->hasTypeTag(BlockTypeTags::MUD) ||
+			$supportBlock->getTypeId() === BlockTypeIds::SOUL_SOIL ||
+			$supportBlock->hasTypeTag(BlockTypeTags::NYLIUM);
```
Comment (ShockedPlot7560) [inline src/block/Fungus.php]: Could we invert the two lign ? So we have all tags before the specific block check.

File: src/block/NetherRoots.php [hunk 15]
```diff
@@ -30,10 +30,12 @@ final class NetherRoots extends Flowable{
...
 
 	private function canBeSupportedAt(Block $block) : bool{
-		//TODO: nylium, moss
+		//TODO: moss
 		$supportBlock = $block->getSide(Facing::DOWN);
 		return
 			$supportBlock->hasTypeTag(BlockTypeTags::DIRT) ||
-			$supportBlock->hasTypeTag(BlockTypeTags::MUD);
+			$supportBlock->hasTypeTag(BlockTypeTags::MUD) ||
+			$supportBlock->getTypeId() === BlockTypeIds::SOUL_SOIL ||
```
Comment (ShockedPlot7560) [inline src/block/NetherRoots.php]: Soul soil is an unrelated change for me.

File: src/block/NetherSprouts.php [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+	use StaticSupportTrait;
+
+	private function canBeSupportedAt(Block $block) : bool{
+		//TODO: moss
+		$supportBlock = $block->getSide(Facing::DOWN);
+		return
+			$supportBlock->hasTypeTag(BlockTypeTags::DIRT) ||
+			$supportBlock->hasTypeTag(BlockTypeTags::MUD) ||
+			$supportBlock->getTypeId() === BlockTypeIds::SOUL_SOIL ||
+			$supportBlock->hasTypeTag(BlockTypeTags::NYLIUM);
```
Comment (ShockedPlot7560) [inline src/block/NetherSprouts.php]: Same here for the order

File: src/block/Netherrack.php [hunk 17]
```diff
@@ -23,9 +23,41 @@
...
+			return true;
+		}
+
+		return false;
+	}
+
+	private function transformToNylium() : bool{
+		foreach($this->getHorizontalSides() as $side){
+			if($side instanceof Nylium){
+				$this->position->getWorld()->setBlock($this->position, $side);
```
Comment (ShockedPlot7560) [inline src/block/Netherrack.php:55]: Why did you use `$side` here, couldn't be better to use a fresh instance ?

File: src/block/Nylium.php [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+	public function onRandomTick() : void{
+		$blockAbove = $this->getSide(Facing::UP);
+		if(!$blockAbove->isTransparent() || $blockAbove->getTypeId() === BlockTypeIds::SNOW_LAYER){
+			//nylium dies
+			BlockEventHelper::spread($this, VanillaBlocks::NETHERRACK(), $this);
+		}
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($this->getSide(Facing::UP)->getTypeId() !== BlockTypeIds::AIR){
```
Comment (ShockedPlot7560) [inline src/block/Nylium.php:60]: You can't fertilize if there is already a fungus or smth like on top of a Nylium block?

File: src/block/Nylium.php (hunk 13 above)
Comment (ShockedPlot7560) [reply src/block/Nylium.php:96]: That's what i see in the wiki. But maybe its different. Wiki says that the block placed when growing is linked to the current biome of the nylium block and not there type ? Maybe the current impl is enough

File: src/block/Netherrack.php (hunk 17 above)
Comment (pandaaaBE) [reply src/block/Netherrack.php:55]: Don´t know what you mean

File: src/block/Nylium.php (hunk 18 above)
Comment (pandaaaBE) [reply src/block/Nylium.php:60]: No? How could that work, by the way, it's still exactly the same code as Grass and they have the same behavior...

File: src/block/Netherrack.php (hunk 17 above)
Comment (pandaaaBE) [reply src/block/Netherrack.php:55]: how should I do it instead? anyways the block state never changes

Comment (pandaaaBE) [reply src/block/Netherrack.php:55]:
```
> private function transformToNylium() : bool{
> 		foreach($this->getHorizontalSides() as $side){
> 			if($side->hasTypeTag(BlockTypeTags::NYLIUM)){
> 				$block = null;
> 				if($side->getTypeId() === BlockTypeIds::CRIMSON_NYLIUM){
> 					$block = VanillaBlocks::CRIMSON_NYLIUM();
> 				}elseif($side->getTypeId() === BlockTypeIds::WARPED_NYLIUM){
> 					$block = VanillaBlocks::WARPED_NYLIUM();
> 				}
> 				
> 				$this->position->getWorld()->setBlock($this->position, $block);
> 				
> 				return true;
> 			}
> 		}
> 
> 		return false;
> 	}
```

	
Do you like this code more?

Comment (ShockedPlot7560) [reply src/block/Netherrack.php:55]: My fault, the original code looks ok

**Review by ShockedPlot7560 — COMMENTED** [review]
Some PHPstan issues need to be adressed

File: tests/phpstan/configs/actual-problems.neon
Comment (ShockedPlot7560) [inline tests/phpstan/configs/actual-problems.neon:1]: Why there is so many strange errors that have been added ?

Comment (pandaaaBE) [reply tests/phpstan/configs/actual-problems.neon:1]: I needed to export `PATH=$PWD/bin/php7/bin:$PATH` and then run the phpstan command again

(thx to dylan)

**Comment by pandaaaBE** [comment]
the only problem left is that the re-generated actual-problems.neon removes following thing:

```
		-
			message: "#^Parameter \\#1 \\$logFile of class pocketmine\\\\utils\\\\MainLogger constructor expects string, string\\|false given\\.$#"
			count: 1
			path: ../../phpunit/scheduler/AsyncPoolTest.php
```

Should I just copy & paste it in the file? lol

**Comment by pandaaaBE** [comment]
Is this pr finished or is something left? I ask because of the Label

File: src/block/VanillaBlocks.php [hunk 19]
```diff
@@ -960,6 +965,16 @@ public function getBreakTime(Item $item) : float{
 		self::register("chiseled_nether_bricks", new Opaque(new BID(Ids::CHISELED_NETHER_BRICKS), "Chiseled Nether Bricks", $netherBrickBreakInfo));
 		self::register("cracked_nether_bricks", new Opaque(new BID(Ids::CRACKED_NETHER_BRICKS), "Cracked Nether Bricks", $netherBrickBreakInfo));
 
+		$nyliumBreakInfo = new Info(BreakInfo::axe(0.4, ToolTier::WOOD, 0.4), [Tags::NYLIUM]);
+		self::register("crimson_nylium", new Nylium(new BID(Ids::CRIMSON_NYLIUM), "Crimson Nylium", $nyliumBreakInfo));
+		self::register("warped_nylium", new Nylium(new BID(Ids::WARPED_NYLIUM), "Warped Nylium", $nyliumBreakInfo));
+
+		$fungusBreakInfo = new Info(BreakInfo::instant());
+		self::register("crimson_fungus", new Fungus(new BID(Ids::CRIMSON_FUNGUS), "Crimson Fungus", $fungusBreakInfo));
+		self::register("warped_fungus", new Fungus(new BID(Ids::WARPED_FUNGUS), "Warped Fungus", $fungusBreakInfo));
```
Comment (ShockedPlot7560) [inline src/block/VanillaBlocks.php]: Need the `POTTABLE_PLANTS` tag. Fungus can be placed in flower pot

File: src/block/Nylium.php [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+			$arr = [
+				VanillaBlocks::WARPED_FUNGUS(),
+				VanillaBlocks::NETHER_SPROUTS(),
+				$roots = VanillaBlocks::WARPED_ROOTS(),
+				$roots,
+				$roots
+			];
+		}
+
+		$random = new Random();
```
Comment (ShockedPlot7560) [inline src/block/Nylium.php]: Why using a random class instead of just `mt_rand` ?

File: src/block/Nylium.php [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+		$count = 8;
+		$radius = 2;
+
+		$arrC = count($arr) - 1;
+		for($c = 0; $c < $count; ++$c){
+			$x = $random->nextRange($this->position->x - $radius, $this->position->x + $radius);
+			$z = $random->nextRange($this->position->z - $radius, $this->position->z + $radius);
+			if($this->position->getWorld()->getBlockAt($x, $this->position->y + 1, $z)->getTypeId() === BlockTypeIds::AIR && $this->position->getWorld()->getBlockAt($x, $this->position->y, $z) instanceof Nylium){
+				$this->position->getWorld()->setBlockAt($x, $this->position->y + 1, $z, $arr[$random->nextRange(0, $arrC)]);
+			}
```
Comment (ShockedPlot7560) [inline src/block/Nylium.php]: It may be possible to reduce the entire line by using the same world variable.

File: src/block/VanillaBlocks.php [hunk 22]
```diff
@@ -960,6 +965,16 @@ public function getBreakTime(Item $item) : float{
...
+
+		self::register("nether_sprouts", new NetherSprouts(new BID(Ids::NETHER_SPROUTS), "Nether Sprouts", $fungusBreakInfo));
```
Comment (ShockedPlot7560) [inline src/block/VanillaBlocks.php]: So, its not really the `fungusBreakInfo`?

**Comment by pandaaaBE** [comment]
Should be ready for a final review

**Review by ShockedPlot7560 — DISMISSED** [review]

**Comment by pandaaaBE** [comment]
Any idea how to resolve this conflicts?

**Comment by ShockedPlot7560** [comment]
Resolve and regenerate the block_factory_consistency
